### PR TITLE
[launch_pytest] Modify how wait_for_output()/wait_for_stderr() work, add assert_*() alternatives

### PR DESCRIPTION
**Pull request by ivanpauno**
Addresses suggestion in https://github.com/ros2/launch/pull/552#discussion_r743890788.

### CODE REVIEW

File: launch_pytest/launch_pytest/tools/process.py [hunk 1]
```diff
@@ -124,6 +104,24 @@ async def wait_for_output(
...
+        except AssertionError:
+            return False
+        return True
+    cond_value = await _wait_for_event_with_condition(
+        launch_context,
+        execute_process_action,
+        _get_stdout_event_handler,
+        condition,
+        timeout)
+    return cond_value if cond_value else validate_output(execute_process_action.get_stdout())
```
Comment (hidmic) [inline launch_pytest/launch_pytest/tools/process.py:122]: @ivanpauno nit: why evaluate the predicate one last time (as opposed to just returning what comes out of `await`)?

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

File: launch_pytest/launch_pytest/tools/process.py (hunk 1 above)
Comment (ivanpauno) [reply launch_pytest/launch_pytest/tools/process.py:122]: to actually throw an assertion error to the user, validate_output() is being wrapped in a condition that catches the AssertionError

**Comment by ivanpauno** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15575/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10260/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13244/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15795/)

File: launch_pytest/launch_pytest/tools/process.py (hunk 1 above)
Comment (hidmic) [reply launch_pytest/launch_pytest/tools/process.py:122]: Ah good point. Why do we `return` though?

Comment (ivanpauno) [reply launch_pytest/launch_pytest/tools/process.py:122]: ah yeah, no need of returning anything here

Comment (ivanpauno) [reply launch_pytest/launch_pytest/tools/process.py]: Fixed in https://github.com/ros2/launch/pull/553/commits/f1abf9fd6b52bbf2405d0bb69b674aa4f80ae8b4.

**Comment by ivanpauno** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15576/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10261/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13245/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15796/)

**Pull request merged**